app/src/main/java/framgia/com/ichat/screen/signup/SignUpContract.java

```diff
@@ -0,0 +1,36 @@
+package framgia.com.ichat.screen.signup;
+
+import android.widget.EditText;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.util.ArrayList;
+
+import framgia.com.ichat.screen.base.BasePresenter;
+
+public class SignUpContract {
+    public interface View {
+        void showProgressDialog();
+
+        void hideProgressDialog();
+
+        void setErrorEmail(String error);
+
+        void setErrorPassword(String error);
+
+        void setErrorEmailEmpty();
+
+        void setErrorPasswordEmpty();
+
+        void requestFocusEmail();
+
+        void showToastShort(String message);
+    }
+
+    public interface Presenter extends BasePresenter<View> {
+        void createAccount(String email, String password, FirebaseAuth auth);
```

Comment:
method create account thì chỉ cần nhập vào username và password
ko có tham số **FirebaseAuth auth**
```suggestion
        void createAccount(String email, String password);
```

FirebaseAuth auth em nên truyền vào thông qua constructor